docs/versioned_docs/version-3.2.0/icicle/primitives/poseidon2.md

```diff
@@ -26,66 +27,74 @@ You can also use your own set of constants as shown [here](https://github.com/in
 
 ### Rust API
 
-This is the most basic way to use the Poseidon2 API.
+This is the most basic way to use the Poseidon2 API. See [examples/poseidon2](examples/rust/poseidon2)
 
 ```rust
-let test_size = 1 << 10;
-let width = 16;
-let rate = 8;
-let ctx = get_default_device_context();
-let poseidon = Poseidon2::load(width, rate, MdsType::Default, DiffusionStrategy::Default, &ctx).unwrap();
+let test_size = 4;
+let poseidon = Poseidon2::new::<F>(test_size,None).unwrap();
 let config = HashConfig::default();
+let inputs = vec![F::one(); test_size];
+let input_slice = HostSlice::from_slice(&inputs);
+//digest is a single element
+let out_init:F = F::zero();
+let mut binding = [out_init];
+let out_init_slice = HostSlice::from_mut_slice(&mut binding);
 
-let inputs = vec![F::one(); test_size * rate as usize];
-let outputs = vec![F::zero(); test_size];
-let mut input_slice = HostOrDeviceSlice::on_host(inputs);
-let mut output_slice = HostOrDeviceSlice::on_host(outputs);
-
-poseidon.hash_many::<F>(
-    &mut input_slice,
-    &mut output_slice,
-    test_size as u32,
-    rate as u32,
-    8, // Output length
-    &config,
-)
-.unwrap();
-```
+poseidon.hash(input_slice, &config, out_init_slice).unwrap();
+println!("computed digest: {:?} ",out_init_slice.as_slice().to_vec()[0]);
 
-In the example above `Poseidon2::load(width, rate, MdsType::Default, DiffusionStrategy::Default, &ctx).unwrap();` is used to load the correct constants based on width and curve. Here, the default MDS matrices and diffusion are used. If you want to get a Plonky3 compliant version, set them to `MdsType::Plonky` and `DiffusionStrategy::Montgomery` respectively.
+```
 
 ## The Tree Builder
 
-Similar to Poseidon1, you can use Poseidon2 in a tree builder.
+Similar to Poseidon1, you can use Poseidon2 in a tree builder. See [examples/poseidon2](examples/rust/poseidon2)
```

Comment:
Broken link as well

Reply:
Leon was supposed to remove poseidon.md and poseidon2.md files from the repo.
Karthik - can you please do that?

Reply:
fixed

Reply:
@danny-shterman Do we want to keep the poseidon.md and poseidon2.md files from rendering? When are planning on adding them back in?

Reply:
@krakhit    Did you rewrite these files? If yes then we should keep themand update them as needed. If not then probably we need to remove them.

Reply:
Ok, poseidon2 is updated and I will add it back to the sidebar so it renders. Poseidon I will leave out of the sidebar so it is not rendered